File: data/multilang_comparison/program_go/program_go/src/func8.go
package main
import "math/rand"
var _ = rand.Intn
func Func8(vars *ArrayParam, PATH0 uint64 , loopsFactor int) *Array{
   var pCounter int = vars.Size;
   _ = pCounter
   if (PATH0 & 1) != 0 {
      var array202 *Array
      if pCounter > 0 {
          array202 = vars.Data[pCounter-1]
          pCounter--
          array202.RefC++
      } else {
          array202 = &Array{
              Size: 796,
              RefC: 1,
              Id: 202,
              Data: make([]uint32, 796),
          }
      }
      var loop59 int = 0;
      var loopLimit59 int = (50)/2 + 1;
      for ; loop59 < loopLimit59; loop59++ {
         for i := 0; i < array202.Size; i++ {
             array202.Data[i]--
         }
         for i := 0; i < array202.Size; i++ {
             if array202.Data[i] == 80 {
                 return array202
             }
         }
      }
      var params0 ArrayParam
      params0.Size = 1
      params0.Data = make([]*Array, params0.Size)
      params0.Data[0] = array202
      var array203 *Array = Func18(&params0, getPath(), loopsFactor);
      params0.Data = nil
      params0.Size = 0
      array203.RefC--
      if array203.RefC == 0 {
          array203.Data = nil
          array203 = nil
      }
      array202.RefC--
      if array202.RefC == 0 {
          array202.Data = nil
          array202 = nil
      }
   
   } else {
      var params0 ArrayParam
      params0.Size = 0
      params0.Data = make([]*Array, params0.Size)
      var array204 *Array = Func14(&params0, getPath(), loopsFactor);
      params0.Data = nil
      params0.Size = 0
      array204.RefC--
      if array204.RefC == 0 {
          array204.Data = nil
          array204 = nil
      }
   }
   var array205 *Array
   if pCounter > 0 {
       array205 = vars.Data[pCounter-1]
       pCounter--
       array205.RefC++
   } else {
       array205 = &Array{
           Size: 812,
           RefC: 1,
           Id: 205,
           Data: make([]uint32, 812),
       }
   }
   return array205;
}

File: data/multilang_comparison/program_cpp_vector/src/func8.cpp
#include "program_cpp_vector.hpp" 
Array* func8(Array_param* vars, const unsigned long PATH0, int loopsFactor) {
   size_t pCounter = vars->size;
   if(PATH0 & 1) {
      Array* array202;
      if (pCounter > 0) {
         array202 = vars->data.at(--pCounter);
         array202->refC++;
         DEBUG_COPY(array202->id);
      } else {
         array202 = new Array();
         array202->size = 796;
         array202->refC = 1;
         array202->id = 202;
         array202->data = std::vector<unsigned int>(array202->size, 0);
         DEBUG_NEW(array202->id);
      }
      unsigned int loop59 = 0;
      unsigned int loopLimit59 = (50)/2 + 1;
      for(; loop59 < loopLimit59; loop59++) {
         for (int i = 0; i < array202->size; i++) {
             array202->data.at(i)--;
         }
         for (int i = 0; i < array202->size; i++) {
             if (array202->data.at(i) == 80) {
                 return array202;
             }
         }
      }
      Array_param params0;
      params0.size = 1;
      std::vector<Array*> data_params0(1);
      params0.data = data_params0;
      params0.data.at(0) = array202;
      Array* array203 = func18(&params0, get_path(), loopsFactor);
      DEBUG_RETURN(array203->id);
      if(params0.data.size() > 0) {
      	params0.data.clear();
      }
      array203->refC--;
      if(array203->refC == 0) {
         array203->data.clear();
         delete array203;
         DEBUG_FREE(array203->id);
      }
      array202->refC--;
      if(array202->refC == 0) {
         array202->data.clear();
         delete array202;
         DEBUG_FREE(array202->id);
      }
   }
   else {
      Array_param params0;
      params0.size = 0;
      std::vector<Array*> data_params0(0);
      params0.data = data_params0;
      Array* array204 = func14(&params0, get_path(), loopsFactor);
      DEBUG_RETURN(array204->id);
      if(params0.data.size() > 0) {
      	params0.data.clear();
      }
      array204->refC--;
      if(array204->refC == 0) {
         array204->data.clear();
         delete array204;
         DEBUG_FREE(array204->id);
      }
   }
   Array* array205;
   if (pCounter > 0) {
      array205 = vars->data.at(--pCounter);
      array205->refC++;
      DEBUG_COPY(array205->id);
   } else {
      array205 = new Array();
      array205->size = 812;
      array205->refC = 1;
      array205->id = 205;
      array205->data = std::vector<unsigned int>(array205->size, 0);
      DEBUG_NEW(array205->id);
   }
   return array205;
}

